### ISSUE TITLE
Bufferlist fills with vim-picker buffers

### ISSUE THREAD

**Issue by boxofrox** [srstevenson/vim-picker#42]
Related to #37 and #39.

I think I belatedly found a side-effect with using `close!` instead of `bdelete!`.  While the picker window still closes consistently, my buffer list is accumulating `term://` entries.

PR to follow.

```
:ls
  1  h   "modules/upp/pages/receiving/js_include.php" line 189
  2  h-  "term://.//22604:git ls-files --cached --exclude-standard --others|fzy ...
  3  h-  "term://.//23334:git ls-files --cached --exclude-standard --others|fzy ...
  4  h-  "term://.//6121:git ls-files --cached --exclude-standard --others|fzy ...
  5  h   "modules/upp/ajax/receiving/receive_uuid.php" line 212
  6  h-  "term://.//11662:git ls-files --cached --exclude-standard --others|fzy ...
  7  h   "modules/phreebooks/functions/phreebooks.php" line 1407
  8  h-  "term://.//16183:echo "term://.//22604:git ls-files --cached ...
 19  h-  "term://.//30248:git ls-files --cached --exclude-standard --others|fzy ...
 22  h-  "term://.//6787:git ls-files --cached --exclude-standard --others|fzy ...
 23  a   "modules/inventory/classes/install.php" line 0
 24  h-  "term://.//28440:git ls-files --cached --exclude-standard --others|fzy ...
 28  h   "[No Name]"                    line 0
 29  h-  "term://.//20669:git ls-files --cached --exclude-standard --others|fzy ...
 30  h-  "term://.//20686:git ls-files --cached --exclude-standard --others|fzy ...
 31 #a   "modules/inventory/config.php" line 0
 32  h-  "term://.//8590:echo "modules/upp/pages/receiving/js_include.php...
 33  h-  "term://.//8783:echo "modules/upp/pages/receiving/js_include.php...
 34  h-  "term://.//9146:echo "modules/upp/pages/receiving/js_include.php...
 35  h-  "term://.//9371:git ls-files --cached --exclude-standard --others|fzy ...
 36  h-  "term://.//9598:echo "modules/upp/pages/receiving/js_include.php...
 37  h-  "term://.//9980:echo "modules/upp/pages/receiving/js_include.php...
 38  h-  "term://.//10630:echo "modules/upp/pages/receiving/js_include.php...
```

**Comment by boxofrox** [comment]
This may be a conflict with other plugins.  A miminal *init.vim* with no plugins loaded doesn't exhibit this behavior.

**Comment by srstevenson** [comment]
Ah, I've worked it out: this happens when [`'hidden'`](https://neovim.io/doc/user/options.html#'hidden') is set. When `'nohidden'` is set, the buffers are deleted when they're hidden. It looks like we will need to temporarily mutate `bufhidden` as suggested in https://github.com/srstevenson/vim-picker/issues/37#issuecomment-451777953:
```viml
let l:bufhidden = &bufhidden
set budhidden=delete
close!
let &bufhidden = l:bufhidden
```

**Comment by boxofrox** [comment]
This bug is weird.  It manifests with `set hidden` used after loading plugins with *vim-plug*, but only with Neovim v0.3.1.  Vim 8.1 is unaffected.

srstevenson's patch does fix the bug I see.

```vim
set nocp
set runtimepath=~/.config/nvim

runtime! ~/.config/nvim/autoload/plug.vim

au!

call plug#begin('~/.config/nvim/plugged')

"Plug 'srstevenson/vim-picker'
Plug '~/files/development/vim/plugins/vim-picker/repo', { 'as': 'vim-picker' }

call plug#end()

runtime! plugin/**/*.vim

set hidden

nmap <leader>lb <Plug>PickerBuffer
nmap <leader>le <Plug>PickerEdit

let g:picker_find_executable = 'fd'
let g:picker_find_flags = '--color=never'
```

**Comment by boxofrox** [comment]
Fixed by #43.
